Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs

```diff
@@ -745,75 +728,124 @@ public Transform MixedRealityPlayspace
-            if (IsInitialized && instance != this)
-            {
-                if (Application.isEditor)
-                {
-                    DestroyImmediate(this);
-                }
-                else
-                {
-                    Destroy(this);
-                }
-
-                Debug.LogWarning("Trying to instantiate a second instance of the Mixed Reality Toolkit. Additional Instance was destroyed");
-            }
-            else if (!IsInitialized)
-            {
-                InitializeInstance();
-            }
-            else
-            {
-                Debug.LogError("Failed to properly initialize the MixedRealityToolkit");
-            }
+            RegisterInstance(this);
         }
 
         private void OnEnable()
         {
-            EnableAllServices();
+            RegisterInstance(this);
+
+            if (IsActiveInstance && Application.isPlaying)
+            {
+                EnableAllServices();
+            }
         }
 
         private void Update()
         {
-            UpdateAllServices();
+            if (IsActiveInstance && Application.isPlaying)
+            {
+                UpdateAllServices();
+            }
         }
 
         private void LateUpdate()
         {
-            LateUpdateAllServices();
+            if (IsActiveInstance && Application.isPlaying)
+            {
+                LateUpdateAllServices();
+            }
         }
 
         private void OnDisable()
         {
-            DisableAllServices();
+            if (IsActiveInstance && Application.isPlaying)
+            {
+                DisableAllServices();
+            }
         }
 
         private void OnDestroy()
         {
-            DestroyAllServices();
-            ClearCoreSystemCache();
+            UnregisterInstance(this);
+        }
 
-            if (instance == this)
-            {
-                instance = null;
-                searchForInstance = true;
+        #endregion MonoBehaviour Implementation
+
+        #region Instance Registration
+
+        private static void RegisterInstance(MixedRealityToolkit toolkitInstance)
+        {
+            if (MixedRealityToolkit.isApplicationQuitting)
+            {   // Don't register instances while application is quitting
+                return;
+            }
+            
+            if (MixedRealityToolkit.activeInstance == null)
+            {
+                // If we don't have an instance, set it here
+                // Set the instance to active
+                MixedRealityToolkit.activeInstance = toolkitInstance;
+                toolkitInstances.Add(toolkitInstance);
+                toolkitInstance.gameObject.SetActive(true);
+                toolkitInstance.InitializeInstance();
+                toolkitInstance.name = ActiveInstanceGameObjectName;
+                return;
+            }
+
+            // If we're already registered, no need to proceed
+            if (!toolkitInstances.Add(toolkitInstance))
```

Comment:
nit: can we please put brackets with our if statements, to follow the coding guidelines? You do this above.

Reply:
HI @Railboy  could you please add the brackets around if statements? Just a reminder.